### DISCUSSION TITLE
CLI Command User security

### DISCUSSION THREAD

**Discussion by leevigraham** [symfony/symfony#54876] [Q&A] [answered]
Is there anyway to "login" a user during a CLI command? 

I know there's no request object so [logging in programatically](https://symfony.com/doc/current/security.html#login-programmatically) doesn't seem to work.

I wonder if there's more low level functions I can use to add a `User` to the current security "session".

**Comment by leevigraham** [comment, marked as answer]
Seems as simple as

```php
$token = new UsernamePasswordToken($user, 'main', $user->getRoles());
$this->tokenStorage->setToken($token);
```

**Reply by philepsybo** [reply]
Thanks for posting this solution. You saved me from a bunch of nasty upcoming head-aches >.<

**Reply by leevigraham** [reply]
This was 2024… might be easier / different now.

**Reply by lon9man** [reply]
@leevigraham so this code will make possible to get current user around cli command (including twig, isGranted and etc) as it works in the web-mode?

### pre-history:
thinking to migrate from laminas to symfony (due to their deprecation roadmap), so investigating functionality currently.
we have background workers, each worker should process business logic using some concrete virtually logged-in user.
